### PR TITLE
Update KFServing docs

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
- Add KFServing feature descriptions along with examples
- Update quick installation with lstio 1.6.2 and KFServing 0.3

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/samples/README.md [hunk 1]
```diff
@@ -1,61 +1,99 @@
...
+| Deploy Transformer with KFServer | [Image Transformer with PyTorch KFServer](./transformer/image_transformer)  |
+| Deploy Transformer with Triton Server | [BERT Model with tokenizer](./triton/bert)  |
+
+### Deploy InferenceService with Explainer and Outlier/Drift Detector
+Model explainability answers the question: "Why did my model make this prediction" for a given instance. KFServing 
+integrates with [Alibi Explainer](https://github.com/SeldonIO/alibi) which implements a black-box algorithm by generating a lot of similar looking intances 
+for a given instance and send out to the model server to produce an explanation.
+Also in order to trust and reliably act on model predictions, it is crucial to monitor the distribution of the incoming
+requests via various different type of detectors. [Alibi Detect](https://github.com/SeldonIO/alibi-detect) checks when the distribution of incoming requests 
+is diverging from a reference distribution such as that of the training data.
```
Comment (cliveseldon) [inline docs/samples/README.md:52]: I think drift and outlier detection should be in a separate section. Also your explanation for them seems to focus on drift not outlier detection which is for individual requests. Maybe reword?

File: docs/samples/README.md [hunk 2]
```diff
@@ -1,61 +1,99 @@
...
+- Batching and Logger
+- Traffic management
+- Security with AuthN/AuthZ
+- Distributed Tracing
+- Out-of-the-box metrics
+- Ingress/Egress control
+
+|   | Exported model| HTTP | gRPC | Examples |
+| ------------- | ------------- | ------------- | ------------- | ------------- |
+| Deploy SKLearn Model on KFServer | pickled model(model.pkl, model.joblib) | :heavy_check_mark: | V2 |[SKLearn Iris](./sklearn)  |
```
Comment (animeshsingh) [inline docs/samples/README.md:22]: Can we move the exported model to the last column? Lot of the values are empty and breaks the readability flow

File: docs/samples/README.md [hunk 3]
```diff
@@ -1,61 +1,99 @@
...
+scale differently from the predictor if your transformer is CPU bound while predictor requires running on GPU. 
+
+| Features  | Examples |
+| ------------- | ------------- |
+| Deploy Transformer with KFServer | [Image Transformer with PyTorch KFServer](./transformer/image_transformer)  |
+| Deploy Transformer with Triton Server | [BERT Model with tokenizer](./triton/bert)  |
+
+### Deploy InferenceService with Explainer and Outlier/Drift Detector
+Model explainability answers the question: "Why did my model make this prediction" for a given instance. KFServing 
+integrates with [Alibi Explainer](https://github.com/SeldonIO/alibi) which implements a black-box algorithm by generating a lot of similar looking intances
```
Comment (animeshsingh) [inline docs/samples/README.md:48]:
```suggestion
integrates with [Alibi Explainer](https://github.com/SeldonIO/alibi) which implements a black-box algorithm by generating a lot of similar looking instances 
```

File: docs/samples/README.md [hunk 4]
```diff
@@ -1,61 +1,99 @@
...
+
+| Features  | Examples |
+| ------------- | ------------- |
+| Deploy Transformer with KFServer | [Image Transformer with PyTorch KFServer](./transformer/image_transformer)  |
+| Deploy Transformer with Triton Server | [BERT Model with tokenizer](./triton/bert)  |
+
+### Deploy InferenceService with Explainer and Outlier/Drift Detector
+Model explainability answers the question: "Why did my model make this prediction" for a given instance. KFServing 
+integrates with [Alibi Explainer](https://github.com/SeldonIO/alibi) which implements a black-box algorithm by generating a lot of similar looking intances 
+for a given instance and send out to the model server to produce an explanation.
```
Comment (animeshsingh) [inline docs/samples/README.md:49]:
```suggestion
for a given instance and sends out to the model server to produce an explanation.
```

File: docs/samples/README.md [hunk 5]
```diff
@@ -1,61 +1,99 @@
...
+### Deploy InferenceService with Cloud/PVC storage
+| Feature  | Examples |
+| ------------- | ------------- |
+| Deploy Model on S3| [Mnist model on S3](./s3) |
+| Deploy Model on PVC| [Models on PVC](./pvc)  |
+| Deploy Model on Azure| [Models on Azure](./azure) |
+
+### Autoscaling
+KFServing's main serverless capability is to allow you to run inference workload without worrying about scaling your service manually once it is deployed. KFServing leverages Knative's [autoscaler](https://knative.dev/docs/serving/configuring-autoscaling/),
+the autoscaler works on GPU as well since the Autoscaler is based on request volume instead of GPU/CPU metrics which can be hard
```
Comment (animeshsingh) [inline docs/samples/README.md:71]:
```suggestion
which works on GPU as well since the autoscaler is based on request volume instead of GPU/CPU metrics which can be hard
```

File: docs/samples/README.md [hunk 6]
```diff
@@ -1,61 +1,99 @@
...
+### Canary Rollout
+Canary deployment enables rollout releases by splitting traffic between different versions to ensure safe rollout.
+
 [Canary Rollout](./rollouts)
 
+### Kubeflow Pipeline Integration
 [InferenceService with Kubeflow Pipeline](./pipelines)
 
-[InferenceService with Request/Response Logger](./logger/basic)
+### Request Batching and Request/Response Logger
```
Comment (animeshsingh) [inline docs/samples/README.md:86]: Are these related in any way? Logger should go much earlier in the list as a supported feature
